plugin-azure-server/src/main/kotlin/jetbrains/buildServer/clouds/azure/arm/KubePodNameGenerator.kt

```diff
@@ -0,0 +1,134 @@
+package jetbrains.buildServer.clouds.azure.arm
+
+import jetbrains.buildServer.clouds.CloudException
+import jetbrains.buildServer.log.Loggers
+import jetbrains.buildServer.serverSide.BuildServerAdapter
+import jetbrains.buildServer.serverSide.BuildServerListener
+import jetbrains.buildServer.serverSide.ServerPaths
+import jetbrains.buildServer.serverSide.executors.ExecutorServices
+import jetbrains.buildServer.util.EventDispatcher
+import jetbrains.buildServer.util.FileUtil
+import jetbrains.buildServer.util.StringUtil
+import org.jetbrains.annotations.NotNull
+import java.io.File
+import java.io.IOException
+import java.nio.charset.Charset
+import java.util.*
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.ReadWriteLock
+import java.util.concurrent.locks.ReentrantReadWriteLock
+
+
+class KubePodNameGenerator(@NotNull eventDispatcher: EventDispatcher<BuildServerListener>,
```

Comment:
I presume this is mostly copy and paste?

Reply:
Actually not. I didn't copy and paste the code because I wanted to write it all out to understand it properly. I believe its just me not formatting properly after unused imports and removing them. 
Extra line will be removed.

Reply:
Should this be `VmNameGenerator` instead?

Reply:
How about we change this to `AzureVmNameGenerator`!

Reply:
Renamed!